### PR TITLE
Booklist delete

### PR DESCRIPTION
**Pull request by clerav**
Besides the view, implemented a reusable and more user/bulma friendly delete confirmation through bulma modal. (We can use this on any delete function. Most of them are currently using browser alerts for confirmation, which looks a bit clunky).

When checking the pull request, please check the url structure of the view, because i think there might be some better alternatives.

### CODE REVIEW

File: BookClub/tests/views/test_delete_booklist_view.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        redirect_url = reverse('booklists_list', kwargs={'username' : 'johndoe'})
+        response = self.client.get(self.url, follow = True)
+        self.assertRedirects(response, redirect_url,
+            status_code=302, target_status_code=200, fetch_redirect_response=True
+        )
+        self.assertTemplateUsed(response, 'user_booklists.html')
+
+        response_messages = messages.get_messages(response.wsgi_request)
+        self.assertEqual(len(response_messages), 1)
+        # below test not working, cannot check message contents...
```
Comment (pleofthegem) [inline BookClub/tests/views/test_delete_booklist_view.py:51]: try something like: 
messages_list = list(response_message.context['messages'])
        self.assertEqual(len(messages_list),1)
        self.assertEqual(str(messages_list[0]) ,'Messages contents here')
        self.assertEqual(messages_list[0].level,messages.ERROR/SUCCESS)

**Review by pleofthegem — APPROVED** [review]
looks good

**Review by jc121F1 — CHANGES_REQUESTED** [review]
Correct message tests as suggested, otherwise good

File: BookClub/tests/views/test_delete_booklist_view.py (hunk 1 above)
Comment (amir-rahim) [reply BookClub/tests/views/test_delete_booklist_view.py]: Fixed now in latest commit.

**Pull request merged**